dhis-2/dhis-web-api/src/main/java/org/hisp/dhis/webapi/utils/FileResourceUtils.java

```diff
@@ -192,8 +193,11 @@ public FileResource saveFileResource( String uid, MultipartFile file, FileResour
 
         File tmpFile = toTempFile( file );
 
+        if ( uid != null && fileResourceService.fileResourceExists( uid ) )
+        {
+            throw new WebMessageException( conflict( format( "A resource with Id %s already exists", uid ) ) );
```

Comment:
Do we want to use an `ErrorCode` here? I know we have many places without but we could try to at least use error codes for new exceptions?